lib/Runtime/Language/JavascriptConversion.cpp

```diff
@@ -1521,4 +1521,37 @@ namespace Js
 
         return NumberUtilities::TryToInt64(length);
     }
+
+    Var JavascriptConversion::TryCanonicalizeAsSimpleVar(Var value)
+    {
+        TypeId typeId = JavascriptOperators::GetTypeId(value);
+        switch (typeId)
+        {
+        case TypeIds_Integer:
+        case TypeIds_Number:
+        {
+            Var taggedInt = TryCanonicalizeAsTaggedInt<true>(value);
+            if (taggedInt)
+            {
+                return taggedInt;
+            }
+
+#if FLOATVAR
+            return value;
+#else
+            return nullptr;
+#endif
+        }
+        case TypeIds_String:
+        case TypeIds_Symbol:
+            // TODO: try to canonicalize Int64/Uint64 to TaggedInt
```

Comment:
This TODO seems like a correctness issue, not an opportunity for future optimization, so I'm leaving a comment here to follow up. In particular, passing a 64-bit int to Has could incorrectly return false because it failed to canonicalize.

Reply:
Good point, I’ll add cases for these.